design/003-interactive-session/003-interactive-session.md

```diff
@@ -0,0 +1,115 @@
+- Start Date: 2021-12-02
+- Status: Proposed
+
+# Run interactive sessions
+
+## Summary
+
+Propose a new sub-group command `renku session` that is responsible for
+managing interactive sessions, both local and remote, for a project.
+
+## Motivation
+
+Renkulab.io provides the option to run interactive sessions of a project. Among other things the session
+contains a jupyter notebook.
+
+The motivation behind this RFC is to provide a simple renku CLI sub-command that basically allows the
+starting and stopping of these interactive sessions.
+
+## Design Detail
+
+The proposed sub-commands of `renku session`, namely to run interactive sessions. The local session heavily relies on
+[docker](https://www.docker.com/). Docker is one of the most popular container engines, that eases the development
+and deployment of applications, by off-loading the installation of the runtime environment dependencies of the
+application.
+
+In order to start a local interactive session, just as in case of renkulab.io, it is required to provide
+a `Dockerfile`. This is by default provided by `renku init` when creating a renku project, but if the user
+is using a custom project template it might not be present.
+
+### Proposed Commands
+
+#### renku session start
+
+A command to start an interactive session.
+
+By default, the `Dockerfile` present in the given project is going to be used for building the docker image
+that's going to be used for running the interactive session. In case the docker image is not available the
+user is going to be prompted whether the user wants to build the docker image, as in some cases it can be
+time consuming.
+
+By using the `--remote [endpoint]` flag an interactive session will be spawned on the specified endpoint. It requires
+that the user has authenticated herself previously with the `renku login` command and aquired the authentication
+token for the endpoint.	If the endpoint is not defined the command will try to assume the endpoint set in the project's
+configuration if available.
```

Comment:
or guess it from the remote?

Reply:
you mean from the `git remote`? in that case i'm wondering in case of multiple remote which one should be used? `origin` or `upstream`?

Reply:
I guess origin

Reply:
deprected... this needs to be addressed in https://github.com/SwissDataScienceCenter/renku-python/issues/2635 but to be honest trying to do this with the git remote is prone to be error.... unless the plugin goes through all the git remote urls and tries to access the notebook api... but for that a `ping` api would be good for the notebook api